plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraCreateAndInsertDataSetup.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.cassandra;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.json.JsonCodec;
+import io.trino.testing.datatype.ColumnSetup;
+import io.trino.testing.datatype.DataSetup;
+import io.trino.testing.sql.SqlExecutor;
+import io.trino.testing.sql.TestTable;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.plugin.cassandra.CassandraMetadata.PRESTO_COMMENT_METADATA;
+import static io.trino.plugin.cassandra.util.CassandraCqlUtils.ID_COLUMN_NAME;
+import static io.trino.plugin.cassandra.util.CassandraCqlUtils.quoteStringLiteral;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+// The reasons for not using CreateAndInsertDataSetup are:
+// (1) Cassandra tables must define a single PRIMARY KEY
+// (2) CQL requires that INSERT INTO statements must provide a list of the columns to be inserted
+public class CassandraCreateAndInsertDataSetup
+        implements DataSetup
+{
+    // The size limit for Cassandra tables is 48 characters
+    // https://cassandra.apache.org/doc/trunk/cassandra/cql/ddl.html#common-definitions
+    private static final Pattern KEYSPACE_AND_TABLE_NAME_PATTERN = Pattern.compile("([a-zA-Z_0-9]{1,48}|\"[a-zA-Z_0-9]{1,48}\")\\.([a-zA-Z_0-9]{1,48}|\"[a-zA-Z_0-9]{1,48}\")");
+    private static final JsonCodec<List<ExtraColumnMetadata>> LIST_EXTRA_COLUMN_METADATA_CODEC = JsonCodec.listJsonCodec(ExtraColumnMetadata.class);
+
+    private final SqlExecutor sqlExecutor;
+    private final String tableNamePrefix;
+    private final CassandraServer cassandraServer;
+
+    public CassandraCreateAndInsertDataSetup(CassandraSqlExecutor sqlExecutor, String tableNamePrefix, CassandraServer cassandraServer)
```

Comment:
Lets' validate if `tableNamePrefix` has a dot and retrieve the keyspace name in constructor.